src/main/java/fact/cleaning/BasicCleaning.java

```diff
@@ -115,52 +106,43 @@
 	 * @param log
 	 * @return
 	 */
-	public ArrayList<Integer> removeStarIslands(ArrayList<Integer> showerPixel, double[] starPosition, PixelSet starSet, double starRadiusInCamera, Logger log) {
+	public PixelSet removeStarIslands(PixelSet showerPixel, double[] starPosition, PixelSet starSet, double starRadiusInCamera, Logger log) {
 
         FactCameraPixel pixel =  pixelMap.getPixelBelowCoordinatesInMM(starPosition[0], starPosition[1]);
         if (pixel == null){
 			log.debug("Star not in camera window. No star islands are removed");
-			return showerPixel;
+			PixelSet pixelSet = new PixelSet();
```

Comment:
why do you have to make a new pixel set and do not return the `showerPixel` set.

Reply:
All other methods of the cleaning return copies and do not modify the pixelSet inplace, so this method should do it also to be consistet.